test_conformance/extensions/cl_khr_external_semaphore/test_external_semaphore_sync_fd.cpp

```diff
@@ -142,16 +100,60 @@ template <bool inorder> cl_int doTest(cl_device_id device, cl_context context)
 // sync fd to a new semaphore
 REGISTER_TEST_VERSION(external_semaphores_import_export_fd, Version(1, 2))
 {
+    REQUIRE_EXTENSION("cl_khr_external_semaphore");
+    REQUIRE_EXTENSION("cl_khr_external_semaphore_sync_fd");
+
+    cl_int err = CL_SUCCESS;
     cl_int total_status = TEST_PASS;
-    cl_int status = doTest<false>(device, context);
-    if (status != TEST_PASS && status != TEST_SKIPPED_ITSELF)
+
+    // test external semaphore sync fd with out-of-order queue
     {
-        total_status = TEST_FAIL;
+        cl_command_queue_properties device_props = 0;
+        err = clGetDeviceInfo(device, CL_DEVICE_QUEUE_PROPERTIES,
+                              sizeof(device_props), &device_props, NULL);
+        test_error(err,
+                   "clGetDeviceInfo for CL_DEVICE_QUEUE_PROPERTIES failed");
+
+        if ((device_props & CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE) == 0)
+        {
+            log_info(
+                "Queue property CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE not "
+                "supported. Skipping test.\n");
+            return TEST_SKIPPED_ITSELF;
```

Comment:
We shouldn't return `TEST_SKIPPED_ITSELF` here because this will skip testing the in-order queue case.